src/highlight.nit

```diff
@@ -432,6 +387,51 @@ class HighlightVisitor
 <script src="http://netdna.bootstrapcdn.com/bootstrap/3.1.1/js/bootstrap.min.js"></script>
 <script>$(".popupable").popover({html:true, placement:'top'})/*initialize bootstrap popover*/</script>"""
 	end
+
+	# Fully process `content` as a Nit source file.
+	#
+	# Set `print_errors = true` to print errors in the code to the console.
+	fun highlightcode(content: String, print_errors: nullable Bool): HLCode
+	do
+		# Prepare a stand-alone tool context
+		var tc = new ToolContext
+		tc.nit_dir = tc.locate_nit_dir # still use the common lib to have core
+		tc.keep_going = true # no exit, obviously
+		if print_errors != true then tc.opt_warn.value = -1 # no output
+
+		# Prepare an stand-alone model and model builder.
+		# Unfortunately, models are enclosing and append-only.
+		# There is no way (yet?) to have a shared module `core` with
+		# isolated and throwable user modules.
+		var model = new Model
+		var mb = new ModelBuilder(model, tc)
+
+		# Parse the code
+		var source = new SourceFile.from_string("", content + "\n")
```

Comment:
I had to do the same thing once, maybe we need a better abstraction for pieces of code that are not source files.

Reply:
A source-file needs a name. https://pm1.narvii.com/6546/9158ae37079f9363f1d9f3346357c9ffe0fa3f38_hq.jpg